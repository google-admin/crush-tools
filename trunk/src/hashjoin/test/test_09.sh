test_number=09
description="multiple input files"

subtest=1
subtest_desc="with header"
infile="$test_dir/input_header.log"
dimfile="$test_dir/dimension_header.log"
outfile="$test_dir/test_$test_number.$subtest.actual"
expected="$test_dir/test_$test_number.$subtest.expected"

$bin -K 'Field-0,Field-1' -J 'Field-2,Field-3' -f $dimfile $infile $infile \
  > "$outfile"

if [ $? -ne 0 ] ||
   [ "`diff -q $outfile $expected`" ]; then
  test_status $test_number $subtest "$description ($subtest_desc)" FAIL
else
  test_status $test_number $subtest "$description ($subtest_desc)" PASS
  rm "$outfile"
fi

subtest=2
subtest_desc="no header"
infile="$test_dir/input_no_header.log"
dimfile="$test_dir/dimension_no_header.log"
outfile="$test_dir/test_$test_number.$subtest.actual"
expected="$test_dir/test_$test_number.$subtest.expected"

$bin -k 1,2 -l 1,2 -j 3,4 -f $dimfile $infile $infile \
  > "$outfile"

if [ $? -ne 0 ] ||
   [ "`diff -q $outfile $expected`" ]; then
  test_status $test_number $subtest "$description ($subtest_desc)" FAIL
else
  test_status $test_number $subtest "$description ($subtest_desc)" PASS
  rm "$outfile"
fi
