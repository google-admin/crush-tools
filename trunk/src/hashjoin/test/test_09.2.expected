1,2,wee,3,4
5,6,w00t,7,8
7,8,nope,,
1,2,wee,3,4
5,6,w00t,7,8
7,8,nope,,
